### ISSUE TITLE
cant install alphalens from pip because of outdated configparser method call

### ISSUE THREAD

**Issue by tinkerdragon** [quantopian/alphalens#406]
(stocks) C:\Users\stanley>pip install alphalens
Collecting alphalens
  Using cached alphalens-0.4.0.tar.gz (24.0 MB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [18 lines of output]
      C:\Users\stanley\AppData\Local\Temp\pip-install-kx2jl_pm\alphalens_7eff4e4e7ff74df7bc217277410482f0\versioneer.py:421: SyntaxWarning: invalid escape sequence '\s'
        LONG_VERSION_PY['git'] = '''
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\stanley\AppData\Local\Temp\pip-install-kx2jl_pm\alphalens_7eff4e4e7ff74df7bc217277410482f0\setup.py", line 35, in <module>
          version=versioneer.get_version(),
                  ^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\stanley\AppData\Local\Temp\pip-install-kx2jl_pm\alphalens_7eff4e4e7ff74df7bc217277410482f0\versioneer.py", line 1480, in get_version
          return get_versions()["version"]
                 ^^^^^^^^^^^^^^
        File "C:\Users\stanley\AppData\Local\Temp\pip-install-kx2jl_pm\alphalens_7eff4e4e7ff74df7bc217277410482f0\versioneer.py", line 1412, in get_versions
          cfg = get_config_from_root(root)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\stanley\AppData\Local\Temp\pip-install-kx2jl_pm\alphalens_7eff4e4e7ff74df7bc217277410482f0\versioneer.py", line 342, in get_config_from_root
          parser = configparser.SafeConfigParser()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      AttributeError: module 'configparser' has no attribute 'SafeConfigParser'. Did you mean: 'RawConfigParser'?
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

(stocks) C:\Users\stanley>python --version
Python 3.12.2

the SafeConfigParser() method has been merged into ConfigParser() method. you guys should update that versioneer.py its from 7 yrs ago!

**Comment by pythonfortraders** [comment]
I was able to get around this by downgrading from 3.12 -> 3.10

**Comment by CharlieBrown-v1** [comment]
Same issues too!

**Comment by wulixin** [comment]
Likely run in watson cloud studio,but can/t install in local coumputer

**Comment by blade-prayer** [comment]
> I was able to get around this by downgrading from 3.12 -> 3.10

Thanks for your suggestion which resolves my problem

**Comment by xingwudao** [comment]
For anyone blocked by current Python installation issues here: I have been building **Ferric Alpha**, a Rust + Polars implementation of alphalens-style factor analysis with a Python API:

https://github.com/xingwudao/ferric-alpha

Docs: https://xingwudao.github.io/ferric-alpha/

It is not a drop-in replacement for this package, but it avoids the old `setup.py` / Versioneer install path and currently covers factor preparation, IC/mean IC, factor weights/returns, alpha/beta, quantile returns, turnover, portfolio series, and native HTML/SVG/PNG tear-sheet rendering.

Sharing here only because this thread is specifically about modern Python installation blockers. If this kind of note is not appropriate for this issue tracker, feel free to remove it.
